### PR TITLE
Fix Javadoc errors in module `client/rest`

### PR DESCRIPTION
**Pull request by gzurowski**
### Description
Fix Javadoc comments to fix Javadoc errors as described in issue https://github.com/opensearch-project/OpenSearch/issues/221.
 
### Issues Resolved
Fixes Javadoc errors in the `client/rest` module.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gzurowski** [comment]
@adnapibar Thanks for your review, I've made the suggested updates.

**Review by setiah — APPROVED** [review]
This looks great @gzurowski . It fixes all 64 errors due to missing javadocs in `client:rest` module, which is pretty awesome!

**Comment by dblock** [comment]
Noticed it was merged into the `javadoc` branch. Was that intentional? This needs to go into `main`, doesn't it? What are the labels we need for this @adnapibar?

**Comment by setiah** [comment]
Yes this was intentional but I agree this needs to go in `main` as well. The problem is, `missingJavadoc` check is only available on `javadoc` branch. It is not added to `main` because of issue #449 . Once that is fixed, we can merge javadoc branch to main (while keeping missingJavadoc checks excluded from `gradlew check`) which will also merge all incoming javadocs. I think we should prioritize the fix as it gets harder to maintain a feature branch. However, until that happens, we have below options 
1. Merge and keep new incoming javadocs in javadoc branch.Fix #449 asap, and then rebase and merge javadoc branch commits into main.
2. Merge new incoming javadocs in javadoc branch and port it to `main` branch as well. (But then it becomes a  backporting nightmare as there are other branches as well)
3. Merge incoming javadocs to main branch (and backport to 1.x as other PRs) and let javadoc branch pull from main. However the problem with merging to main is there is no check available to validate the new javadocs. The checks are only available in javadoc branch. So if we go with this, contributor has to first verify against javadoc branch, then push to main and 1.x branch. (javadoc branch can be maintained by maintainers)
@dblock i'm fine with both 1 & 3, but 3 seems to have slightly more friction. WDYT?

**Comment by nknize** [comment]
This is the second time I've seen this precommit failure:

```
* What went wrong:
A problem was found with the configuration of task ':client:benchmark:validateNebulaPom' (type 'PomValidationTask').
> File '/var/CITOOL/workflow/OpenSearch_CI/PR_Checks/Gradle_Precommit/search/client/benchmark/build/publications/nebula/pom-default.xml' specified for property 'pomFile' does not exist.
```

We shouldn't get into a habit of merging w/ failing precommit. Is this a CI config issue? Red herring?

**Comment by setiah** [comment]
> This is the second time I've seen this precommit failure:
> 
> ```
> * What went wrong:
> A problem was found with the configuration of task ':client:benchmark:validateNebulaPom' (type 'PomValidationTask').
> > File '/var/CITOOL/workflow/OpenSearch_CI/PR_Checks/Gradle_Precommit/search/client/benchmark/build/publications/nebula/pom-default.xml' specified for property 'pomFile' does not exist.
> ```
> 
> We shouldn't get into a habit of merging w/ failing precommit. Is this a CI config issue? Red herring?

@nknize the precommit fails because of #449 . That's why we are keeping these changes in separate feature (javadoc) branch, until we fix that. Good news though, I have figured a fix for #449 , and once that's merged, the precommit should start passing and then we can merge javadoc branch to main

**Pull request merged**
